pkg/registry/core/pod/strategy.go

```diff
@@ -249,9 +249,12 @@ var EphemeralContainersStrategy = podEphemeralContainersStrategy{Strategy}
 
 // dropNonEphemeralContainerUpdates discards all changes except for pod.Spec.EphemeralContainers and certain metadata
 func dropNonEphemeralContainerUpdates(newPod, oldPod *api.Pod) *api.Pod {
-	pod := dropPodUpdates(newPod, oldPod)
-	pod.Spec.EphemeralContainers = newPod.Spec.EphemeralContainers
-	return pod
+	newEphemeralContainerSpec := newPod.Spec.EphemeralContainers
+	newPod.Spec = oldPod.Spec
+	newPod.Status = oldPod.Status
+	metav1.ResetObjectMetaForStatus(&newPod.ObjectMeta, &oldPod.ObjectMeta)
```

Comment:
also need to implement podEphemeralContainersStrategy#GetResetFields to indicate the fields reset by strategy so that managedFields are calculated correctly ... not sure if that means enumerating all the *other* fields under spec as being reset or not

```
			fieldpath.MakePathOrDie("metadata"),
			fieldpath.MakePathOrDie("spec", ...),
			fieldpath.MakePathOrDie("status"),
```

cc @jpbetz for input here

Reply:
oh, I think we'd use the same `resetFieldsFilter: fieldpath.NewIncludeMatcherFilter...` approach the resize strategy used, and filter out everything but `spec.ephemeralContainers`

Reply:
> oh, I think we'd use the same `resetFieldsFilter: fieldpath.NewIncludeMatcherFilter...` approach the resize strategy used, and filter out everything but `spec.ephemeralContainers`

Yes, exactly.  We introduced that last release and it allows for this level of masking.

Reply:
thanks, added `resetFieldsFilter` and `GetResetFieldsFilter` to `podEphemeralContainersStrategy`